lib/checkers.h

```diff
@@ -31,9 +31,11 @@ enum class ReportType : std::uint8_t {
     autosar = 1,
     certC = 2,
     certCpp = 3,
-    misraC = 4,
-    misraCpp2008 = 5,
-    misraCpp2023 = 6,
+    misraC2012 = 4,
+    misraC2023 = 5,
+    misraC2025 = 6,
+    misraCpp2008 = 7,
```

Comment:
I think it's better that misraCpp2008 and misraCpp2023 will keep the indexes. Otherwise users can get a different report type set after upgrade.